### DISCUSSION TITLE
Help with syncing MiniCAM and Open Ephys Acquisition Board!

### DISCUSSION THREAD

**Discussion by mateusvillarroel** [bonsai-rx/bonsai#2037] [Q&A]
Hello! Our group is currently trying to synchronize OE acquisition with a MiniCAM recording. Both systems are operational within their GUIs, however, I'm having a hard time assembling a workflow that enables simultaneous recording. The main problem is that when I run the workflow with both sequences separated - the OE one is the most common example and the MiniCAM one is the one available in Aharoni's repository -, the video recorded is sped up, although it maintains quality. Even though I've used the I/O board to connect the "Sync Output" or "Trigger Input" on the Miniscope DAQ to the digital input or output on the Acquisition Board, I really have no clue on how to link both sequences on Bonsai!

I suspect that the video length is shorter because the camera can't really keep up with the board acquisition frame. However, when I disable the Rhd2000EvalBoard sequence and run the worflow, the problem keeps ocurring. I can confirm through OE GUI that the TTL signal flows adequately, but I don't understand how can I input this ping onto the other system through Bonsai... 

Any help will be appreciated!

**Comment by bruno-f-cruz** [comment]
Have you confirmed that the video sped up is real and not the result of the encoding? In other words, if you ignore the duration of the video, and focus on the number of frames, does it make sense?

**Reply by mateusvillarroel** [reply]
Yes, I have! When I check the .csv dataframe generated with the "frame number" MemberSelector, it shows that I have about ~6 FPS. This is the case for when I set the "FPS" property to 50 in the MiniCAM node, and also 50 FPS in the VideoWriter node. When I record it at 10 or 40 FPS, which are valid options in the MiniCAM workflow, there is a frame loss, but it is similar to the one we get when record on its own GUI and the videos don't seem so sped up. I'll try to get hold of the video example and its frame count.

**Reply by mateusvillarroel** [reply]
Better late than never! Here I provide .csv files for the TTLs generated by the camera and the framelist, as well as a Google Drive link for the corresponding videosample, that should've been around 8 second-long. Let me know what you think! 

[minicam-frame-timestamp-ccm619-pt2024-10-16T10_46_50.csv](https://github.com/user-attachments/files/17564746/minicam-frame-timestamp-ccm619-pt2024-10-16T10_46_50.csv)

[ttl_signal2024-10-16T10_46_50.csv](https://github.com/user-attachments/files/17564748/ttl_signal2024-10-16T10_46_50.csv)

https://drive.google.com/file/d/1kXbgbaVgCj61ibS4Aw1fObJNhcXCYWCG/view?usp=sharing
